build_tooling/requirements-compatibility-py311.txt

```diff
@@ -0,0 +1,2 @@
+# Makes sure we are able to use Numpy 1
+numpy<2
```

Comment:
So what matrix of Python (minor) and Pandas/numpy (major) versions are now covered by the CI?

Reply:
The following is the matrix on Linux based on the latest run in this branch:

| Python Version      | NumPy Version | Pandas Version |
|---------------------|---------------|----------------|
| Python 3.7          | 1.21.6        | 1.3.5          |
| Python 3.8          | 1.24.4        | 2.0.3          |
| Python 3.8 (Compat) | 1.24.4        | 1.5.3          |
| Python 3.9          | 2.0.2         | 2.2.3          |
| Python 3.10         | 2.2.1         | 2.2.3          |
| Python 3.11         | 2.2.1         | 2.2.3          |
| Python 3.11 (Compat)| 1.26.4        | 2.2.3          |
| Python 3.12         | 2.2.1         | 2.2.3          |

There is no Numpy 2 for python 3.8 and below.
Python 3.6 backwards/forwards compat can still be tested when the persistent tests are run.